ocean_lib/data_provider/base.py

```diff
@@ -248,13 +248,30 @@ def write_file(
         else:
             logger.warning(f"consume failed: {response.reason}")
 
+    @staticmethod
+    @enforce_types
+    def _sanitize_content(header: str) -> bool:
+        pattern = re.compile(r"\\|\.\.|/")
+        if pattern.findall(header):
+            return False
+        return True
+
     @staticmethod
     @enforce_types
     def _get_file_name(response: Response) -> Optional[str]:
         try:
-            return re.match(
-                r"attachment;filename=(.+)", response.headers.get("content-disposition")
-            )[1]
+            if DataServiceProviderBase._sanitize_content(
```

Comment:
I don't understand. Are we validating or sanitizing? It seems we are validating, so please rename this function _validate_content_disposition and return True if valid, False otherwise.

Even more, we can have some generic header validation, e.g. _validate_headers(response), which for now only validates the content-disposition header.

I also suggest treating the denied case first, e.g.
```
if not _validate_content_disposition(response.headers.get("content-disposition"):
     logger.error(...)
     return None
return re.match(r"attachment ...)
```